bioptim/dynamics/problem.py

```diff
@@ -465,7 +511,16 @@ def configure_tau(nlp, as_states: bool, as_controls: bool):
         if as_states:
             nlp.x = vertcat(nlp.x, all_tau[0])
             nlp.var_states["tau"] = nlp.shape["tau"]
-            # Add plot if it happens
+
+            tau_bounds = nlp.x_bounds[nlp.shape["q"] + nlp.shape["qdot"]: nlp.shape["tau"]]  # tau as a state with q
```

Comment:
I think you meant:
`nlp.shape["q"] + nlp.shape["qdot"]: nlp.shape["q"] + nlp.shape["qdot"] + nlp.shape["tau"]`

Reply:
note that the line after, you are creating `nq_qdot` which could be useful here:
`nq_qdot: nq_qdot + nlp.shape["tau"]`